collector/exporter.go

```diff
@@ -142,21 +148,27 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
 	e.totalScrapes.Inc()
 	var err error
-
-	scrapeTime := time.Now()
-	db, err := sql.Open("mysql", e.dsn)
-	if err != nil {
-		log.Errorln("Error opening connection to database:", err)
-		e.error.Set(1)
-		return
+	var wg sync.WaitGroup
+	if atomic.LoadInt32(&inited) == 0 {
+		mtx.Lock()
+		defer mtx.Unlock()
+		if atomic.LoadInt32(&inited) == 0 {
+			db, err = sql.Open("mysql", e.dsn)
+			if err != nil {
+				log.Errorln("Error opening connection to database:", err)
+				e.error.Set(1)
+				return
+			}
+			atomic.StoreInt32(&inited, 1)
+			maxCon := e.collect.MaxMySQLConns
+			if maxCon > 16 {
```

Comment:
The limit comes from a command line parameter. The max limit I wanted to make was 64, but put 16 in the code. That's a miss. We limit the number connections so that the exporter doesn't create performance issues for other MySQL users, by limiting the number of parallel query the exporter can fire (that's equal to max connection pool size). Generally DBA puts a limit on max number of connections that can be used by an user from server side. We are doing that from Client side also, to be nice.

Reply:
Makes sense. Maybe move the magic number to a named const and explain it in a comment than?

Reply:
Yes. Will do that.